service/src/main/java/uk/gov/hmcts/reform/fpl/service/UserService.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.idam.client.IdamApi;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+/**
+ * Gets Users from Idam.
+ */
+@Service
+public class UserService {
+
+    private final IdamApi idamApi;
+
+    @Autowired
+    public UserService(IdamApi idamApi) {
+        this.idamApi = idamApi;
+    }
+
+    /**
+     * Makes a request to Idam and returns the user details for the user who is making the request.
+     *
+     * @param authorization the authorization token generated by the callback.
+     * @return caseLocalAuthority gained from the getEmailDomain function.
+     */
+    public String getUserDetails(String authorization) {
+        UserDetails userDetails = idamApi.retrieveUserDetails(authorization);
+        String email = userDetails.getEmail();
+        return getEmailDomain(email);
+    }
+
+    /**
+     * Removes all characters from an email address apart from the domain.
```

Comment:
I am not a fun of JavaDoc on private methods. IMO private methods should be so small that method name should describe exactly what it does. For me it is a bit unnecessary effort that does not give much value.

I am OK with JavaDoc on public methods as it sometimes helps using the API. Are we as. a team happy writing JavaDoc for every single public method?

Reply:
Yep, happy to write docs for the public methods